### PR TITLE
Admin UI: Encountering infinite loading when entering non-existent identity/client/tier ids in URL

### PR DESCRIPTION
**Pull request by stamenione**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

Button  "Find" is now disabled if the input field is empty. I found that problem when testing this endpoint "/api/v1/Identities/{identityAddress}/DeletionProcesses/AuditLogs", and the test case was the empty string that could be entered.

image

image

This is the Dark mode version of error screen.
image

The Light version.
image

### CODE REVIEW

File: Applications/AdminUi/apps/admin_ui/lib/home/client_details/client_details.dart [hunk 1]
```diff
@@ -80,12 +81,21 @@ class _ClientDetailsState extends State<ClientDetails> {
...
+          _clientDetails = clientDetails.data;
+          _selectedTier = _clientDetails!.defaultTier;
+        });
+      }
+    } catch (e) {
+      final errorMessage = e.toString().split(':')[2].trim();
+
+      if (mounted) {
+        context.go('/error', extra: errorMessage);
+      }
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/home/client_details/client_details.dart:98]: we use results to NOT have to use try..catch - please respect that.

File: Applications/AdminUi/apps/admin_ui/lib/screens/error_screen.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            ),
+            Gaps.h16,
+            Text(
+              errorMessage,
+              textAlign: TextAlign.center,
+              style: TextStyle(fontSize: 18, color: Theme.of(context).colorScheme.tertiary, decoration: TextDecoration.none),
+            ),
+            Gaps.h16,
+            FilledButton(
+              onPressed: () => context.go('/identities'),
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/error_screen.dart:45]: go to home goes back to identities when you originated at tiers?

File: Applications/AdminUi/apps/admin_ui/lib/home/client_details/client_details.dart (hunk 1 above)
Comment (jkoenig134) [reply Applications/AdminUi/apps/admin_ui/lib/home/client_details/client_details.dart:98]: this is also scattered everywhere in this PR and has to be fixed everywhere.

**Comment by jkoenig134** [comment]
The review for this will take a bit of time b/c I am not totally sure if this is the correct approach. With `context.go("/error")` you override the whole navigation stack, instead to replace the top most page for example. I am pretty sure there is a way better option for this (e.g. pushReplacement or something similar).

**Comment by stamenione** [comment]
> The review for this will take a bit of time b/c I am not totally sure if this is the correct approach. With `context.go("/error")` you override the whole navigation stack, instead to replace the top most page for example. I am pretty sure there is a way better option for this (e.g. pushReplacement or something similar).

I see with pushReplacement it eliminates the use of queryParameters and then the returnRoute that I added is completely unnecessary. Then the the only option that I can think of is to pass the errorMessage to the ErrorScreen through an extra parameter but I am not entirely sure if that is correct way.

File: Applications/AdminUi/apps/admin_ui/lib/screens/error_screen.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            Gaps.h16,
+            Text(
+              errorMessage,
+              textAlign: TextAlign.center,
+              style: TextStyle(fontSize: 18, color: Theme.of(context).colorScheme.tertiary, decoration: TextDecoration.none),
+            ),
+            Gaps.h16,
+            FilledButton(
+              onPressed: () => context.pop(),
+              child: Text(context.l10n.errorScreen_goToHome, style: TextStyle(fontSize: 16, color: Theme.of(context).colorScheme.onPrimary)),
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/error_screen.dart:49]: setting the color should not be necessary, please try it without.

File: Applications/AdminUi/apps/admin_ui/lib/home/deletion_process_details/deletion_process_details.dart
[the remark this answers is not in the document]
Comment (jkoenig134) [reply Applications/AdminUi/apps/admin_ui/lib/home/deletion_process_details/deletion_process_details.dart:1]: so these were unused?

[the remark this answers is not in the document]
Comment (stamenione) [reply Applications/AdminUi/apps/admin_ui/lib/home/deletion_process_details/deletion_process_details.dart:1]: Yes

[the remark this answers is not in the document]
Comment (jkoenig134) [reply Applications/AdminUi/apps/admin_ui/lib/home/deletion_process_details/deletion_process_details.dart:1]: hm, then I don't really understand why this is related to this PR..

[the remark this answers is not in the document]
Comment (stamenione) [reply Applications/AdminUi/apps/admin_ui/lib/home/deletion_process_details/deletion_process_details.dart:1]: I can revert those changes and create a separate PR for this change.

[the remark this answers is not in the document]
Comment (jkoenig134) [reply Applications/AdminUi/apps/admin_ui/lib/home/deletion_process_details/deletion_process_details.dart:1]: No need. Now it's included and the confusion is already resolved :)

**Comment by tnotheis** [comment]
A manual test was successful. @jkoenig134 If you don't have anything to add, I'm okay with merging it.

**Review by tnotheis — APPROVED** [review]

**Comment by jkoenig134** [comment]
Looks fine for me.

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**